extensions/objects/templates/FilepathEditor.js

```diff
@@ -31,109 +31,118 @@ oppia.directive('filepathEditor', [
       scope: true,
       template: '<div ng-include="getTemplateUrl()"></div>',
       controller: ['$scope', function($scope) {
-        // Reset the component each time the value changes (e.g. if this is part
-        // of an editable list).
+        // Reset the component each time the value changes
+        // (e.g. if this is part of an editable list).
         $scope.$watch('$parent.value', function(newValue) {
-          $scope.localValue = {
-            label: newValue || ''
-          };
-          $scope.imageUploaderIsActive = false;
+          if (newValue) {
+            $scope.setSavedImageUrl(newValue, false);
+          }
         });
 
-        $scope.explorationId = explorationContextService.getExplorationId();
+        $scope.validate = function(state) {
+          return state.state === 'saved' &&
+                 state.savedUrl &&
+                 state.savedUrl.length > 0;
+        };
 
-        $scope.validate = function(localValue) {
-          return localValue.label && localValue.label.length > 0;
+        $scope.isNoImageUploaded = function() {
+          return !$scope.state || $scope.state.state === 'empty';
```

Comment:
Consider defining named constants $scope.MODE_EMPTY etc. and using those, since you use these strings in multiple places.